### PR TITLE
[CI] Truncate commit message to be at most 50,000 characters 

### PR DESCRIPTION
**Pull request by weiwangmeta**
To avoid "docker command too long errors" 
e.g. nightly failures https://github.com/pytorch/pytorch/actions/runs/4523340751/jobs/7966584446

In these actions, the -e COMMIT_MESSAGES would be supplied to "docker" command. 
E.g. as of today (03/26/2023), the output of git cherry -v origin/master from nightly branch is about 141761 characters. Because, each day, the following number of characters would be added to the COMMIT_MESSAGE (the nightly branch started from 2019-11-08)
+ c9ed10c3773297c1f9c041fce95156e25a198388 2023-03-26 nightly release (dc45ad702466e4a73d972d3e1dc0c12ed80d2eef)

### CODE REVIEW

File: .circleci/verbatim-sources/job-specs/job-specs-custom.yml [hunk 1]
```diff
@@ -250,6 +250,9 @@
             # then trim all special characters like single and double quotes to avoid unescaped inputs to
             # wreak havoc internally
             export COMMIT_MESSAGES="${COMMIT_MESSAGES//[\'\"]}"
+            # then truncate the message length to 10,000 characters, not doing so will make docker command
+            # fail due to nightly vs master difference
+            export COMMIT_MESSAGES="${COMMIT_MESSAGES:0:10000}"
```
[the remark this answers is not in the document]
Comment (weiwangmeta) [reply .circleci/verbatim-sources/job-specs/job-specs-custom.yml:255]: Is not the nightly branch the only branch that produce 10K characters? I can also bump it up to a high number to make it the case. 
For majority if not all branches, this is a no-op.  
What benefit does it provide if the truncated 10,000 character message complete or not?

File: .github/workflows/_mac-test.yml [hunk 2]
```diff
@@ -177,6 +177,9 @@ jobs:
           # then trim all special characters like single and double quotes to avoid unescaped inputs to
           # wreak havoc internally
           export COMMIT_MESSAGES="${COMMIT_MESSAGES//[\'\"]}"
+          # then truncate the message length to 10,000 characters, not doing so will make docker command
+          # fail due to nightly vs master difference
+          export COMMIT_MESSAGES="${COMMIT_MESSAGES:0:10000}"
```
[the remark this answers is not in the document]
Comment (weiwangmeta) [reply .github/workflows/_mac-test.yml:182]: Ok, I will undo the update.

File: .github/workflows/_win-test.yml [hunk 3]
```diff
@@ -190,6 +190,9 @@ jobs:
           # then trim all special characters like single and double quotes to avoid unescaped inputs to
           # wreak havoc internally
           export COMMIT_MESSAGES="${COMMIT_MESSAGES//[\'\"]}"
+          # then truncate the message length to 10,000 characters, not doing so will make docker command
+          # fail due to nightly vs master difference
+          export COMMIT_MESSAGES="${COMMIT_MESSAGES:0:10000}"
```
[the remark this answers is not in the document]
Comment (weiwangmeta) [reply .github/workflows/_win-test.yml:195]: I am not seeing such failures on Windows, so I will undo these changes too. Thanks for flagging!

File: .circleci/config.yml [hunk 4]
```diff
@@ -725,6 +725,9 @@ jobs:
             # then trim all special characters like single and double quotes to avoid unescaped inputs to
             # wreak havoc internally
             export COMMIT_MESSAGES="${COMMIT_MESSAGES//[\'\"]}"
```
[the remark this answers is not in the document]
Comment (weiwangmeta) [reply .circleci/config.yml]: I ended up only changing _linux-test.yml as that is the only platform that I'm observing those kind of issues. 
Removed the export part.

File: .circleci/verbatim-sources/job-specs/job-specs-custom.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (weiwangmeta) [reply .circleci/verbatim-sources/job-specs/job-specs-custom.yml]: Increased the 10,000 to 50,000 characters

File: .github/workflows/_mac-test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (weiwangmeta) [reply .github/workflows/_mac-test.yml]: Only changing _linux-test.yml now.

File: .github/workflows/_win-test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (weiwangmeta) [reply .github/workflows/_win-test.yml]: Ditto.

**Comment by kit1980** [comment]
Talked to the team, this should no longer be needed.

**Pull request closed without merging**